### DISCUSSION TITLE
firewall ports for direct connection

### DISCUSSION THREAD

**Discussion by jasonafriedman** [jamesog/tailscale-edgeos#28] [Q&A]
I am trying to get a direct connection to work. Right now, it connects using DERP and is slower than a direct connection. Enabling UPnP works, but that is insecure. Here are the the relevant docs:

[https://tailscale.com/kb/1181/firewalls/](url)
[https://tailscale.com/kb/1082/firewall-ports/](url)

Tailscale needs the following firewall rules:

- Let your internal devices initiate TCP connections to *:443
- Let your internal devices initiate UDP from :41641 to *:*
- Let your internal devices initiate UDP to *:3478

Do you know which commands to use for this? I tried forwarding ports using the GUI but really don't know what I'm doing.

**Comment by jamesog** [comment]
Unless you're restricting what connections can be made outbound, you shouldn't need to do anything. It's hard to say without knowing more about your setup.
